sample_size/multiple_testing.py

```diff
@@ -0,0 +1,57 @@
+from typing import List
+
+import numpy as np
+from statsmodels.stats.multitest import multipletests
+
+from sample_size.metrics import BaseMetric
+
+
+class MultipleTestingMixin:
+    """
+    This class calculates sample size required under the case of multiple testing
+
+    Attributes:
+    metrics: a list of BaseMetric registered by users
+    variants: number of variants, including control
+    alpha: statistical significance
+    power: statistical power
+
+    """
+
+    metrics: List[BaseMetric]
+    alpha: float
+    power: float
+    variants: int
+
+    def get_multiple_sample_size(self, lower: float, upper: float, depth: int = 0) -> int:
+        MAX_RECURSION_DEPTH = 20
+        EPSILON = 0.025
+
+        if depth > MAX_RECURSION_DEPTH:
+            raise RecursionError
+
+        candidate = int((upper + lower) / 2)
+        expected_power = self._expected_average_power(candidate)
+
+        if np.isclose((self.power, expected_power), EPSILON) or np.isclose((candidate, upper), EPSILON):
+            return candidate
```

Comment:
By definition, if we're evaluating the latter condition, we know `candidate` is not close to the true sample size, so we shouldn't return it. Technically, you don't need to perform this calculation at all (the recursion depth will implicitly handle this case), but if you would like to end the recursion early, you can add this second condition.

```suggestion
        if np.isclose((self.power, expected_power), EPSILON):
            return candidate
       
        if upper == lower:
            raise RuntimeError("f"Couldn't find a sample size that satisfies the power you requested: {self.power}"
```

Reply:
I think that concern would be best addressed by whatever client is calling this function. It should probably be sufficient to just raise the exception